src/moscot/problems/generic/_generic.py

```diff
@@ -233,7 +236,7 @@ def prepare(
                 raise TypeError("`x_attr` and `y_attr` must be of type `str` or `dict`.")
 
         xy, kwargs = handle_joint_attr(joint_attr, kwargs)
-        xy, x, y = handle_cost(xy=xy, x=GW_updated[0], y=GW_updated[1], cost=cost)
+        xy, x, y = handle_cost(xy=xy, x=GW_updated[0], y=GW_updated[1], cost=cost, cost_kwargs=cost_kwargs)
```

Comment:
In the above code, there's a bug: `[{}] * 2` are pointers, try
```python
x = [{}] * 2
x[0]['a'] = 42
print(x)
```
Please fix using `[{}, {}]` or better yet, define a local function which does:
```python
def set_quad_defaults(cont: Dict[str, Any) -> Dict[str, Any]:
  ...

xy, kwargs = handle_joint_attr(joint_attr, kwargs)
x = set_quad_defaults(x)
y = set_quad_defaults(y)
xy, x, y = handle_cost(xy=xy, x=x, y=y, cost=cost, cost_kwargs=cost_kwargs)
```